src/directory/mmap_directory.rs

```diff
@@ -367,22 +340,17 @@ impl Directory for MmapDirectory {
     /// removed before the file is deleted.
     fn delete(&self, path: &Path) -> result::Result<(), DeleteError> {
         let full_path = self.resolve_path(path);
-        match fs::remove_file(&full_path) {
-            Ok(_) => self.sync_directory().map_err(|e| DeleteError::IoError {
-                io_error: e,
-                filepath: path.to_path_buf(),
-            }),
```

Comment:
removed, because we rely on the os now?

Reply:
No... Removed because we call sync_directory a single time after deleting all of the file, instead of doing it after deleting every single file.

Reply:
see the change in managed_directory l.195